aquadoggo/src/db/stores/test_utils.rs

```diff
@@ -380,10 +392,97 @@ pub fn test_db(
 /// Container for `SqlStore` with access to the document ids and key_pairs used in the
 /// pre-populated database for testing.
 pub struct TestDatabase {
+    pub context: Context,
     pub store: SqlStorage,
     pub test_data: TestData,
 }
 
+impl TestDatabase {
+    /// Publish a document and materialise it in the store.
+    ///
+    /// Also runs dependency task for document.
+    pub async fn add_document(
```

Comment:
These are really cool methods, I like how they work. 

I might like to also explore a flow where we use existing `send_to_store()` or db config with `populate()` functions for inserting entries and operations and then a new function called `materialise()` which just materialises everything in the store. I like the pattern of _not_ adding any new methods to the db store and instead modifying with helpers. I think this gives nice separation between what is possible with the store alone and what is extended test functionality.

For `add_schema` there could be  a helper function which takes a store and fields and then inserts the needed operations internally. Quite similar to what's here, just not on the test db.

I'm very happy to raise this is an issue and discus further though.

Reply:
I first tried implementing this using the `Node` for p2panda but then I was missing the schema store there so that's how I ended up with this approach. I also didn't want the materialisation to happen async so that the tests don't need to wait for some magical millisecond number (that then causes failing tests when run on a very slow system).

> I like the pattern of not adding any new methods to the db store and instead modifying with helpers

Do you mean adding new methods in the sense of "adding methods to the test store that don't exist on the production store"? That seems like a good idea, yes. I mostly wanted to have a single-function interface for adding schemas and documents based on some shorthand spec. We can also move that to functions instead of methods on the store! Maybe as a follow-up?

Reply:
> Do you mean adding new methods in the sense of "adding methods to the test store that don't exist on the production store"?

Yeh! This is what I meant :+1:

Reply:
I'll do a follow-up and convert the added methods to functions!

https://github.com/p2panda/aquadoggo/issues/226